editor-components/feature/feature12/feature12.tsx

```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
   }
 
   render() {
+    const upperTitle = this.getPropValue("upperTitle");
+    const title = this.getPropValue("title");
+    const titleBackingText = this.getPropValue("titleBackingText");
+    const itemCount = this.getPropValue("itemCount");
+    const messageCards = this.castToObject<IMessages[]>("messageCard");
+    const firstItem = this.getPropValue("message");
+    const firstCardTitle = firstItem[0].value;
+    const buttonText = firstItem[1].value;
+    const backgroundImage = firstItem[2].value;
+   
     return (
-      <div className={this.decorateCSS("container")}>
-        <div className={this.decorateCSS("max-content")}>
+      <div className={this.decorateCSS("container")} style={{ paddingBottom: "0px" }}>
+        <div className={this.decorateCSS("upper-title")} style={{ fontSize: "1.2em", textAlign: "center" }}>
+          {upperTitle}
+        </div>
+        <div className={this.decorateCSS("shadow-header-title")} style={{ fontSize: "12.5em", textAlign: "center", marginBottom: "20px" }}>
+          {titleBackingText}
+        </div>
+        <div className={this.decorateCSS("header-title")} style={{ fontSize: "5em", textAlign: "center", marginTop: "35px" }}>
+          {title}
+        </div>
+        <div className={this.decorateCSS("max-content")} style={{ marginTop: "20px" }}>
           <div className={this.decorateCSS("content")}>
-            {this.castToObject<IMessages[]>("message-card").map(
-              (message: any, index: number) => (
-                <div className={this.decorateCSS("card-item-count")} style={{
-                  width: 90 / this.getPropValue("itemCount") + "%",
-                }}>
-                <div className={this.decorateCSS("message")} key={index}>
-                  <h3 className={this.decorateCSS("title")}>{message.title}</h3>
-                  <p className={this.decorateCSS("long-text")}>{message.description}</p>
-                </div>
-                </div>
-              )
+            {firstItem.length > 0 && (
```

Comment:
firstItem = prop "message" is an object. it doesn't have length. it is not an array.
it is an object so we should create a type for it.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-approach-props)